### PR TITLE
Lk pd2249

### PR DESCRIPTION
**Pull request by ekiernan**
### Description

This work adds verification tasks to compare the h5ad files that result from Optimus and Multiome (Optimus and ATAC workflows). The GEX verification is separate from the ATAC verification because the ATAC h5ad does not actually contain a matrix or a variables, just observations. 

----

### Checklist 
If you can answer "yes" to the following items, please add a checkmark next to the appropriate checklist item(s) **and** notify our WARP documentation team by tagging either @ekiernan or @kayleemathews in a comment on this PR.

- [ ] Did you add inputs, outputs, or tasks to a workflow?
- [ ] Did you modify, delete or move: file paths, file names, input names, output names, or task names?
- [ ] If you made a changelog update, did you update the pipeline version number?

### CODE REVIEW

File: verification/VerifyTasks.wdl [hunk 1]
```diff
@@ -331,42 +331,107 @@ task CompareLooms {
...
     File truth_h5ad
     File test_h5ad
-
-    Int cpu = 1
-    String docker = "gcr.io/gcp-runtimes/ubuntu_16_0_4:latest"
-    Int disk_size_gb = ceil(size(truth_h5ad, "GiB") + size(test_h5ad, "GiB")) + 20
-    Int memory_mb = ceil(size(truth_h5ad, "MiB") + size(test_h5ad, "MiB") * 2)
+    String docker = "python:3.10.0-buster"
+    Int disk_size_gb = ceil(size(truth_h5ad, "GiB") + size(test_h5ad, "GiB")) + 200
+    Int memory_gb = 20
```
Comment (khajoue2) [inline verification/VerifyTasks.wdl:340]: does it help to use a more standard machine memory? 16 or 32

File: verification/VerifyTasks.wdl [hunk 2]
```diff
@@ -331,42 +331,107 @@ task CompareLooms {
...
+    truth = ad.read_h5ad(truth_h5ad)
+    test = ad.read_h5ad(test_h5ad)
+    
+    truth_obs = pd.DataFrame(truth.obs)
+    test_obs = pd.DataFrame(test.obs)
+    
+    print("Now running obs equivalence check")
+    
+    if truth_obs.equals(test_obs)==True:
+        print("true")
```
Comment (khajoue2) [inline verification/VerifyTasks.wdl:366]: Could we print "pass" or a different message here? Is the "true" value used somewhere else?

File: verification/VerifyTasks.wdl [hunk 3]
```diff
@@ -331,42 +331,107 @@ task CompareLooms {
...
-      echo "H5ad file sizes are NOT identical"
-      exit 1
-    fi
+task CompareH5adFilesGEX {
+  input {
+    File truth_h5ad
+    File test_h5ad
+    String docker = "python:3.10.0-buster"
+    Int disk_size_gb = ceil(size(truth_h5ad, "GiB") + size(test_h5ad, "GiB")) + 200
+    Int memory_mb = ceil(size(truth_h5ad, "MiB") + size(test_h5ad, "MiB") * 5)
```
Comment (khajoue2) [inline verification/VerifyTasks.wdl:389]: We should fix memory

File: verification/VerifyTasks.wdl [hunk 4]
```diff
@@ -331,42 +331,107 @@ task CompareLooms {
...
+    truth_var = pd.DataFrame(truth.var)
+    test_var = pd.DataFrame(test.var)
+    
+    truth_sum = truth.X.sum()
+    test_sum = test.X.sum()
+    
+    print("Now running equivalence check")
+    
+    if truth_obs.equals(test_obs)==True and truth_var.equals(test_var)==True and truth_sum==test_sum:
+        print("true")
```
Comment (khajoue2) [inline verification/VerifyTasks.wdl:421]: same comment here

**Review by khajoue2 — CHANGES_REQUESTED** [review]

File: verification/VerifyTasks.wdl (hunk 3 above)
Comment (ekiernan) [reply verification/VerifyTasks.wdl:389]: @khajoue2 Thanks for taking a look. For this file comparison, the dynamic memory worked. What do you suggest changing?

**Review by nikellepetrillo — APPROVED** [review]
looks good!

**Review by khajoue2 — APPROVED** [review]

**Pull request merged**